validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -31,176 +31,316 @@
 LOGGER = logging.getLogger(__name__)
 
 
-class BatchSubmitFinisher(Handler):
-    def __init__(self, block_store, batch_cache):
+class _ClientRequestHandler(Handler):
+    """Parent class for all Client Request Handlers.
+
+    Handles the repetitive tasks of parsing a client request, and formatting
+    the response. Also includes some helper methods common to multiple client
+    handlers. Child classes should not implement the `handle` method directly,
+    instead defining a unique `_respond` method specified below.
+
+    Args:
+        request_proto (class): Protobuf class of the request to be handled
+        response_proto (class): Protobuf class of the response to be sent
+        response_type (enum): Message status of the response
+        tree (MerkleDatabase, optional): State tree to be queried
+        block_store (BlockStoreAdapter, optional): Block chain to be queried
+        batch_cache (TimedCache, optional): A cache of Batches being processed
+
+    Attributes:
+        _status (class): Convenience ref to response_proto for accessing enums
```

Comment:
I think I actually want to keep this. Our linter allows inherited classes to access `_` methods, but will complain if any outside code tries to. As `handle` is the only method I want used outside of this module, that is exactly the sort of behavior I'm looking for.

Since there doesn't seem a strong style preference in the repo to break that protection by writing inherited methods without an `_`, I think it should stay in this case.